### DISCUSSION TITLE
Latency with Rockwell PLC

### DISCUSSION THREAD

**Discussion by alattanzi** [eclipse-milo/milo#1271] [General]
Hello all,

I create an OPC client to comunicate with Siemens and Rockwell PLC.
In the first case, the average reading time 85 tags is around 100/150 ms
In the second, the average reading time the same number of tags is greater of 800 ms. With rockwell i try to read only one tag, and the avg time is around 20ms

Anyone had the same issue? How to fix it?
The instruction used to read the tags is:
_List<DataValue> results = opc.readValues(0, TimestampsToReturn.Server, nodes).get();_

Thanks
Andrea

**Comment by kevinherron** [comment]
I'm not sure this is an issue with the client. 

I'm pretty sure Logix PLCs de-prioritize OPC UA traffic in favor of CIP traffic, so if there's already a high comms load on this PLC this may be as good as it gets. In any case it doesn't look like you're doing anything wrong on your side of this. 

If you get a Wireshark capture you can probably confirm the latency is just waiting for the PLC to respond.

**Comment by kevinherron** [comment]
If you are repeatedly reading these values you should consider using a Subscription, as this is more efficient than repeat reads.

**Comment by alattanzi** [comment]
Thanks for the reply.
For Sure I'm going to implement also the Subscription in my application.
Is there a way to change the priority of traffic OPC/CIP? If yes, is this in the PLC or in the OPC server?

Thanks

**Reply by kevinherron** [reply]
I don't know. It would be in the PLC / OPC UA server, not something you could do from the client.

**Reply by pturmel** [reply]
It is documented to prioritize CIP unconditionally.
